avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1215,7 +1224,12 @@ private Schema toConnectSchema(org.apache.avro.Schema schema, boolean forceOptio
         throw new DataException("Invalid schema name: " + connectNameJson.toString());
       }
       name = connectNameJson.asText();
-
+    } else if (schema.getLogicalType() != null) {
+      String logicalTypeName = schema.getLogicalType().getName();
+      name = AVRO_LOGICAL_TYPE_NAMES_TO_CONNECT_LOGICAL_TYPE_NAMES.get(logicalTypeName);
+      if (name == Decimal.LOGICAL_NAME) {
```

Comment:
This is going to get messy as we add more logical types (and doesn't lend itself to extension if we decide to support that). For the data, we introduce a small interface and cleanly separate the conversion to/from logical type and allow that to handle fields like the `scale` on `Decimal`. Could we do something similar here?